### PR TITLE
Feature/kotlin

### PR DESCRIPTION
**Pull request by Sanduhr32**
Fixes #14.

Changes proposed in this pull request:
-

- More Delegates
- Many fixes and optimization again
- And other stuff

### CODE REVIEW

File: src/main/java/ml/duncte123/skybot/objects/command/Command.java [hunk 1]
```diff
@@ -51,38 +49,48 @@
...
-
+    
     /**
      * Reloads the list of people who have upvoted this bot
      */
     protected static void reloadUpvoted() {
+        if (cooldown && Settings.useCooldown) return;
         try {
-            String token = AirUtils.config.getString("apis.discordbots_userToken", "");
+            String token = AirUtils.config.getString("apis.discordbots_userToken");
```
Comment (duncte123) [inline src/main/java/ml/duncte123/skybot/objects/command/Command.java]: Erm, why did you remove the default value?

File: src/main/java/ml/duncte123/skybot/utils/ConfigUtils.java [hunk 2]
```diff
@@ -35,7 +35,7 @@
     public ConfigUtils() {
         try {
             AirUtils.log(Level.INFO, "Loading config.json");
-            this.config = ConfigLoader.getConfig(new File("config.json"));
+            this.config = ConfigLoader.getConfig(new File(".\\config.json"));
```
Comment (duncte123) [inline src/main/java/ml/duncte123/skybot/utils/ConfigUtils.java]: doesn't that do the same as just using `new File("config.json")`?

File: src/main/java/ml/duncte123/skybot/utils/Settings.java [hunk 3]
```diff
@@ -45,37 +40,42 @@
...
     /**
      * The icon url for the embeds
      */
-    public static final String defaultIcon = "https://bot.duncte123.me/img/favicon.png";
-
+    public static final String defaultIcon = "https://dshelmondgames.ml/favicon";
     /**
      * The colour of the bar that your embed has
      */
-    public static final Color defaultColour = decode(AirUtils.config.getString("discord.embedColour", "#0751c6"));
```
Comment (duncte123) [inline src/main/java/ml/duncte123/skybot/utils/Settings.java]: plz fix this file

Comment (Sanduhr32) [reply src/main/java/ml/duncte123/skybot/utils/Settings.java]: Idk how that happened

File: src/main/java/ml/duncte123/skybot/utils/ConfigUtils.java (hunk 2 above)
Comment (Sanduhr32) [reply src/main/java/ml/duncte123/skybot/utils/ConfigUtils.java]: Not important both works

File: src/main/java/ml/duncte123/skybot/objects/command/Command.java (hunk 1 above)
Comment (Sanduhr32) [reply src/main/java/ml/duncte123/skybot/objects/command/Command.java]: I think it went rip while GSON to JSON rewrite

File: src/main/java/ml/duncte123/skybot/commands/essentials/eval/EvalCommand.java [hunk 4]
```diff
@@ -192,10 +194,12 @@ public void executeCommand(String invoke, String[] args, GuildMessageReceivedEve
...
                 sendMsg(event, "ERROR: " + e3.getClass().getName() + ": " + e3.getMessage());
                 sendError(event.getMessage());
+                // Debuging System.out.println(EarthUtils.throwableToJSONObject(e3).toString(4));
+            } catch (ArrayIndexOutOfBoundsException e4) {
+                sendSuccess(event.getMessage());
             }
         } catch (Throwable thr) {
-            sendMsg(event, "ERROR: " + thr.toString());
-            thr.printStackTrace();
+            sendErrorJSON(event.getMessage(), thr, true);
```
Comment (duncte123) [inline src/main/java/ml/duncte123/skybot/commands/essentials/eval/EvalCommand.java]: I don't want json files for error messages

File: src/main/java/ml/duncte123/skybot/utils/Settings.java [hunk 5]
```diff
@@ -45,37 +38,42 @@
...
+    /**
+     * This is the title from the player embed
+     */
+    public static final String playerTitle = "AirPlayer";
     /**
      * The icon url for the embeds
      */
-    public static final String defaultIcon = "https://bot.duncte123.me/img/favicon.png";
-
+    public static final String defaultIcon = "https://dshelmondgames.ml/favicon";
```
Comment (duncte123) [inline src/main/java/ml/duncte123/skybot/utils/Settings.java]: Wrong icon

File: src/main/java/ml/duncte123/skybot/utils/Settings.java [hunk 6]
```diff
@@ -45,37 +38,42 @@
...
-
     /**
      * This is the name that your bot has
      */
     public static final String defaultName = "DuncteBot";
-
+    /**
+     * This is the title from the player embed
+     */
+    public static final String playerTitle = "AirPlayer";
```
Comment (duncte123) [inline src/main/java/ml/duncte123/skybot/utils/Settings.java]: This has been moved

File: src/main/java/ml/duncte123/skybot/utils/Settings.java (hunk 5 above)
Comment (Sanduhr32) [reply src/main/java/ml/duncte123/skybot/utils/Settings.java]: Fix it yourself i just pulled it from dev...

File: src/main/java/ml/duncte123/skybot/commands/essentials/eval/EvalCommand.java (hunk 4 above)
Comment (Sanduhr32) [reply src/main/java/ml/duncte123/skybot/commands/essentials/eval/EvalCommand.java]: k

File: src/main/java/ml/duncte123/skybot/utils/Settings.java (hunk 6 above)
Comment (Sanduhr32) [reply src/main/java/ml/duncte123/skybot/utils/Settings.java]: tell me where

Comment (duncte123) [reply src/main/java/ml/duncte123/skybot/utils/Settings.java]: AudioUtils

File: src/main/java/ml/duncte123/skybot/utils/Settings.java (hunk 5 above)
Comment (duncte123) [reply src/main/java/ml/duncte123/skybot/utils/Settings.java]: fine

File: src/main/java/ml/duncte123/skybot/utils/Settings.java (hunk 6 above)
Comment (Sanduhr32) [reply src/main/java/ml/duncte123/skybot/utils/Settings.java]: thanks

File: src/main/java/ml/duncte123/skybot/utils/Settings.java (hunk 5 above)
Comment (Sanduhr32) [reply src/main/java/ml/duncte123/skybot/utils/Settings.java]: i did it..

Comment (duncte123) [reply src/main/java/ml/duncte123/skybot/utils/Settings.java]: ty ❤️

File: src/main/java/ml/duncte123/skybot/commands/essentials/eval/EvalCommand.java (hunk 4 above)
Comment (Sanduhr32) [reply src/main/java/ml/duncte123/skybot/commands/essentials/eval/EvalCommand.java]: uh thats the just in case catch... idk whats better so tell me what you want or i add a boolean in Settings

Comment (duncte123) [reply src/main/java/ml/duncte123/skybot/commands/essentials/eval/EvalCommand.java]: make a boolean

**Comment by Sanduhr32** [comment]
@duncte123 since which release is this in [dev](https://github.com/duncte123/SkyBot/tree/dev)?

**Comment by duncte123** [comment]
@Sanduhr32 I don't know

**Comment by Sanduhr32** [comment]
Found it [3.51.5](https://github.com/duncte123/SkyBot/releases/tag/v3.51.5)
